src/commonControls/listControl/ListControlAPL.ts

```diff
@@ -115,3 +116,234 @@ export namespace ListControlAPLPropsBuiltIns {
+                    },
+                ],
+                items: [
+                    {
+                        type: 'Sequence',
+                        scrollDirection: 'vertical',
+                        data: '${listItems}',
+                        width: '100%',
+                        height: '100%',
+                        paddingLeft: '0',
+                        numbered: true,
+                        items: [
+                            {
+                                type: 'Container',
+                                items: [
+                                    {
+                                        type: 'AlexaTextListItem',
+                                        touchForward: true,
+                                        hideOrdinal: false,
+                                        disabled: '${disableScreen}',
+                                        primaryText: '${data.primaryText}',
+                                        primaryAction: {
+                                            type: 'Sequential',
+                                            commands: [
+                                                {
+                                                    type: 'SendEvent',
+                                                    arguments: ['${controlId}', '${ordinal}'],
+                                                },
+                                                {
+                                                    type: 'SetValue',
+                                                    componentId: 'root',
+                                                    property: 'disableScreen',
+                                                    value: true,
+                                                },
+                                                {
+                                                    type: 'SetValue',
+                                                    componentId: 'root',
+                                                    property: 'debugText',
+                                                    value: 'Done Selected',
+                                                },
+                                            ],
+                                        },
+                                    },
+                                ],
+                            },
+                        ],
+                    },
+                ],
+            });
+
+            const itemIds: string[] = control.getListItemIDs(input);
+            // Create the inline document, which instantiates the Layout
+            const listItems = itemIds.map((x) => ({
+                primaryText: props.valueRenderer!(x, input).primaryText,
+            }));
+
+            return {
+                type: 'ListControl-touchForward',
+                controlId: control.id,
+                listItems,
+            };
+        } else if (props.renderStyle === 'imageList') {
```

Comment:
refactor:  move the separate implementations to separate methods, e.g. renderImageList(...)